src/dataviews/dataviews-factory.js

```diff
@@ -69,12 +81,17 @@ module.exports = Model.extend({
   },
 
   _indexOf: function (layerModel) {
-    var index = this._interactiveLayersCollection.indexOf(layerModel);
+    // We need to filter the layers to only select those that Windshaft knows about
+    // and be able to calculate the right index.
+    var interactiveLayers = this._layersCollection.select(function (layer) {
+      return layer.get('type') === 'CartoDB' || layer.get('type') === 'torque';
+    });
+
+    var index = interactiveLayers.indexOf(layerModel);
```

Comment:
After talking: this can indeed be removed, but better to do it in a separate PR. Especially, due to other requirements with vector integration, that will move the ownership of filters to a layer instead. This will also have the nice side-effect of cleaning up filters collection and windshaft-map (serialized layer should include filters that instead of the custom `FiltersCollection.toJSON`).